cmd/bee/cmd/deploy.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/node"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func (c *command) initDeployCmd() error {
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Deploy and fund the chequebook contract",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if (len(args)) > 0 {
+				return cmd.Help()
+			}
+
+			var logger logging.Logger
+			switch v := strings.ToLower(c.config.GetString(optionNameVerbosity)); v {
+			case "0", "silent":
+				logger = logging.New(ioutil.Discard, 0)
+			case "1", "error":
+				logger = logging.New(cmd.OutOrStdout(), logrus.ErrorLevel)
+			case "2", "warn":
+				logger = logging.New(cmd.OutOrStdout(), logrus.WarnLevel)
+			case "3", "info":
+				logger = logging.New(cmd.OutOrStdout(), logrus.InfoLevel)
+			case "4", "debug":
+				logger = logging.New(cmd.OutOrStdout(), logrus.DebugLevel)
+			case "5", "trace":
+				logger = logging.New(cmd.OutOrStdout(), logrus.TraceLevel)
+			default:
+				return fmt.Errorf("unknown verbosity level %q", v)
+			}
+
+			dataDir := c.config.GetString(optionNameDataDir)
+			factoryAddress := c.config.GetString(optionNameSwapFactoryAddress)
+			swapInitialDeposit := c.config.GetString(optionNameSwapInitialDeposit)
+			swapEndpoint := c.config.GetString(optionNameSwapEndpoint)
+
+			stateStore, err := node.InitStateStore(logger, dataDir)
+			if err != nil {
+				return err
+			}
+
+			signerConfig, err := c.configureSigner(cmd, logger)
+			if err != nil {
+				return err
+			}
+			signer := signerConfig.signer
+
+			ctx := cmd.Context()
+
+			swapBackend, overlayEthAddress, chainID, transactionService, err := node.InitChain(
+				ctx,
+				logger,
+				stateStore,
+				swapEndpoint,
+				signer,
+			)
+			if err != nil {
+				return err
+			}
+
+			chequebookFactory, err := node.InitChequebookFactory(
+				logger,
+				swapBackend,
+				chainID,
+				transactionService,
+				factoryAddress,
+			)
+			if err != nil {
+				return err
+			}
+
+			if _, err = node.InitChequebookService(
+				ctx,
+				logger,
+				stateStore,
+				signer,
+				chainID,
+				swapBackend,
+				overlayEthAddress,
+				transactionService,
+				chequebookFactory,
+				swapInitialDeposit,
+			); err != nil {
+				return err
+			}
+
+			return nil
```

Comment:
you can drop the last if and just return the err from `InitChequebookService `

Reply:
Fixed redundant if.